lib/vdsm/storage/lvmdevices.py

```diff
@@ -65,8 +66,12 @@ def configure(vgs):
     # On the other hand configuring correct devices file doesn't cause any
     # harm.
     try:
-        _create_system_devices(vgs)
-    except cmdutils.Error:
+        if vgs:
+            _create_system_devices(vgs)
+        else:
+            # Create empty devices file.
+            Path(_get_devices_file_path()).touch()
+    except (cmdutils.Error, lvmconf.UnexpectedLvmConfigOutput):
```

Comment:
First issue:
This code calls `_get_devices_file_path()`, which calls `lvmconf.configured_value`, that can raise both `cmdutils.Error` (not exactly know where, but is caught in mainline code https://github.com/oVirt/vdsm/blob/master/lib/vdsm/storage/lvmdevices.py#L103) and `lvmconf.UnexpectedLvmConfigOutput`.

Second issue:
`_create_system_devices` didn't raise ` lvmconf.UnexpectedLvmConfigOutput`, it is only required due to the addition of the `_get_devices_file_path` cal.
This block is creating the devices file, either empty or with some device, and if it fails, we handle exceptions consistently, in both cases, log warning and raise. I don't see how this is unrelated, if I split the new exception into a new commit we create a "wrong" commit with inconsistent behavior for `lvmconf.UnexpectedLvmConfigOutput`. But in any case, since the only different is a warning logged that would probably not be even printed in the tool, I could remove the exception from the except clause.

Third issue:
It is a simple standard function, with a very descriptive name for what it does. Makes a nice one-liner. And is not even new to the standard, been there since Python 3.4 iirc. But sure, I'll look for whatever other way to create empty files we have in the project.

About creating files using lvm, I experimented a bit with that. `lvmdevices --update` creates an empty devices file. However, the bug describes installing hypervisor in partitions without `lvm`, so I thought that creating an empty file directly would ensure that this solution always works.

Reply:
We always have the lvm package - it is required by vdsm.

Looking at the current code, I don't know why we log the warning in line 70 before
we raise. Logging and raising is bad anti-pattern that unfortunately is common in
vdsm. The best way to handle errors around _create_system_devices() is the same way
we handle other errors - do not handle anything and let the error propagate to the caller.

I think what should be done is:
- remove the useless warning and try block around the call to `_create_system_devices()`
- fix it to always create a devices file,  using way lvm provides if possible

Reply:
Unfortunately, `lvm` does not provide a reliable way, thus touching a file is the simplest way.

Anyway, as discussed, I added a comment in the file, generated with the same format and vibes as LVM would.